### ISSUE TITLE
Adding ACH (US bank account debit) into the UPE

### ISSUE THREAD

**Issue by jessepearson** [Automattic/woocommerce-payments#3506]
We are moving forward with adding ACH (US bank account debit) into the UPE, full spike outline here: pcqRLn-rw-p2

This is presently only available on our Dev account at Stripe, which you must be logged in to in order to view the doc:
https://stripe.com/docs/payments/us-bank-account/accept-a-payment

Notes from the spike:
* We need to have a `customer_id` to send to Stripe when creating the intent on the checkout page. This should be able to be done by using the user’s customer_id if they are logged in, or their session info if they are not logged in. [Doc reference](https://stripe.com/docs/payments/us-bank-account/accept-a-payment?platform=web#web-create-payment-intent).
* The docs then say this: “Stripe.js then loads an on-page modal UI that handles bank account details collection and verification.” [Doc reference](https://stripe.com/docs/payments/us-bank-account/accept-a-payment?platform=web#web-collect-details).
  * I am unsure if this means that the modal will appear once “Bank” is chosen or if the order must be submitted. With Sepa, the field for the bank info appears within the UPE before the order is submitted. There is a screenshot below with what I am seeing now.
  * Also unsure if the modal referred to here is additional inline fields within the UPE, or if there’s an actual pop-up.
  * If the order must be submitted, we may need to override the behavior of the *Place order* button.
* The data collected in the modal also gives the customer an option for instant verification or not.
  * If instant verification is not chosen, then micro deposits are made to the account, and then an email is sent from Stripe to the customer to verify the amounts. This gives another failure point in the process, due to if they are not able to properly verify the account, the transaction fails on Stripe’s end. [Doc reference](https://stripe.com/docs/payments/us-bank-account/accept-a-payment?platform=web#web-verify-with-microdeposits).
  * Instant verification comes with a legal binding for us: https://stripe.com/en-ca/legal/instant-verification-services
  * It is also possible to default to [instant only](https://stripe.com/docs/payments/us-bank-account/accept-a-payment?platform=web#instant-only-verification) or [micro deposit only](https://stripe.com/docs/payments/us-bank-account/accept-a-payment?platform=web#microdeposit-only-verification) verification.
* Once the data is collected, we then must display a mandate confirming the customer gives permission to debit their account, which is similar to SEPA. Then the payment can be submitted.
* Due to this is a delayed payment method, we must put the order on-hold, and then wait for a webhook to move it to processing. This can be hours or days. [Doc reference](https://stripe.com/docs/payments/us-bank-account/accept-a-payment?platform=web#web-confirm-paymentintent-succeeded).

**Comment by jessepearson** [comment]
After a conversation with Stripe, I am putting this on hold for now. Even using their example code, which is outside of WCPay scope, we get an error. They are looking into it and will be getting back to us: p1638197967188200-slack-C9976E5MJ

There was also team discussion, so I wanted to include that for relevance: p1639485324076300-slack-C01B8KNUYSW

**Comment by jessepearson** [comment]
Stripe was able to fix the issue on their end with this. This is now blocked by https://github.com/Automattic/woocommerce-payments/issues/3448 . Without the Payment Element being updated, the bank method's fields do not show.

**Comment by jessepearson** [comment]
As it turns out, Stripe was having API issues while I was working on this and https://github.com/Automattic/woocommerce-payments/issues/3448 is actually not a blocker. I created a PR to fix the issue, and have tested with and without the code in my ACH PR and there is now no change, they both work. Holy frustrating, Batman.

**Comment by jessepearson** [comment]
Placing on hold per: paJDYF-3dw-p2#comment-10360

**Comment by BlackWireKevin** [comment]
Any update on getting this done?

**Comment by csmcneill** [comment]
> Any update on getting this done?

We do not have an update at this time. Adding ACH to [WooCommerce Payments](https://woocommerce.com/products/woocommerce-payments/?aff=10486&cid=1131038) is still on the roadmap, but we do not currently have an ETA.

**Comment by jessepearson** [comment]
I am going to close this issue for now. The research performed in order to add the payment method is now 2 years old, meaning everything here is outdated. When ACH is added, the team handling it will need to open a new issue with a new research spike, etc.
